### PR TITLE
feat(erc20): allow permissionless registration of erc20

### LINKED ISSUE #37: allow for permissionless erc20 token pair registration

### PR DESCRIPTION
**Pull request by mtsitrin**
closes #37

### CODE REVIEW

File: x/erc20/keeper/proposals.go [hunk 1]
```diff
@@ -88,6 +88,12 @@ func (k Keeper) RegisterERC20(
 		)
 	}
 
+	if len(metadata.DenomUnits) != 2 {
```
Comment (mtsitrin) [inline x/erc20/keeper/proposals.go:91]: added additional enforcement, so erc20 that wants to be registered must provide symbol name and decimals

File: x/erc20/types/params.go [hunk 2]
```diff
@@ -18,29 +18,36 @@ package types
...
 
 // Parameter store key
 var (
-	ParamStoreKeyEnableErc20   = []byte("EnableErc20")
-	ParamStoreKeyEnableEVMHook = []byte("EnableEVMHook")
+	ParamStoreKeyEnableErc20     = []byte("EnableErc20")
+	ParamStoreKeyEnableEVMHook   = []byte("EnableEVMHook")
+	ParamStoreKeyRegistrationFee = []byte("RegistrationFee")
+
+	DefaultRegistrationFee = math.NewInt(10).MulRaw(1e18) // 10 tokens of native denom
```
Comment (mtsitrin) [inline x/erc20/types/params.go:31]: @omritoptix FYI

**Review by VictorTrustyDev — APPROVED** [review]
LGTM

**Review by keruch — APPROVED** [review]
GJ!

**Pull request merged**